src/PublicApiGeneratorTests/NullableTests.cs

```diff
@@ -0,0 +1,603 @@
+           public System.Collections.Generic.Dictionary<int?, string>? GetSecretData4(System.Collections.Generic.Dictionary<int?, string>? value) { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Structs()
+        {
+            AssertPublicApi<Structs>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Structs
+    {
+           public System.Collections.Generic.KeyValuePair<string?, int?> field;
+           public Structs() { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Tuples()
+        {
+            AssertPublicApi<Tuples>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Tuples
+    {
+           public Tuples() { }
+           public System.Tuple<string, string?, int, int?> Tuple1(System.Tuple<string, string?, int, int?>? tuple) { }
+           public System.ValueTuple<string, string?, int, int?> Tuple2(System.ValueTuple<string, string?, int, int?>? tuple) { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Constraints()
+        {
+            AssertPublicApi<Constraints>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Constraints
+    {
+           public Constraints() { }
+           public void Print1<T>(T val)
+               where T : class { }
+           public void Print2<T>(T val)
+               where T : class? { }
+           public static void Print3<T>()
+                where T : System.IO.Stream { }
+           public static void Print4<T>()
+                where T : System.IDisposable { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Nullable_Constraints()
+        {
+            AssertPublicApi(typeof(Constraints2<,>),
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Constraints2<X, Y>
+        where X : System.IComparable<X>
+        where Y : class?
+    {
+           public Constraints2() { }
+           public T Convert<T>(T data)
+               where T : System.IComparable<string?> { }
+           public static void Print1<T>()
+                where T : System.IO.Stream? { }
+           public static void Print2<T>()
+                where T : System.IDisposable? { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_BaseType()
+        {
+            AssertPublicApi<NullableComparable>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class NullableComparable : System.Collections.Generic.List<string?>, System.IComparable<string?>
+    {
+           public NullableComparable() { }
+           public int CompareTo(string? other) { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_OpenGeneric()
+        {
+            AssertPublicApi(typeof(StringNullableList<>),
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class StringNullableList<T> : System.Collections.Generic.List<T?>
+        where T : struct
```

Comment:
Isn't the `where T : class` case the interesting one?

Reply:
done